internal/build/job/manager.go

```diff
@@ -52,6 +58,27 @@ func (jbm *jobManager) Sync(ctx context.Context, mod kmmv1beta1.Module, m kmmv1b
 
 	logger.Info("Building in-cluster")
 
+	// if theres no build specified skip
+	if !module.ShouldBeBuilt(&mod, &m) {
```

Comment:
I would prefer if ShouldBeBuild, ShouldBeSigned and checkImageExists will not be in the Sync function.  There are other flows ( like preflight), that has done image verification in a different way, and just want to execute the Build flow, without running the checks again. WDYT about moving those verification to a separate API? Then this API can be called prior to calling Sync in a regular flow, and not called in the flow that does not need it. Or add and API that call those checks and then calls the Sync API

Reply:
great, thank you!